Sources/GRPC/ClientCalls/ClientCall.swift

```diff
@@ -42,14 +43,15 @@ public protocol ClientCall {
   var status: EventLoopFuture<GRPCStatus> { get }
 
   /// Trailing response metadata.
-  var trailingMetadata: EventLoopFuture<HTTPHeaders> { get }
+  var trailingMetadata: EventLoopFuture<HPACKHeaders> { get }
 
   /// Cancel the current call.
   ///
   /// Closes the HTTP/2 stream once it becomes available. Additional writes to the channel will be ignored.
   /// Any unfulfilled promises will be failed with a cancelled status (excepting `status` which will be
   /// succeeded, if not already succeeded).
-  func cancel()
+  func cancel() -> EventLoopFuture<Void>
+  func cancel(promise: EventLoopPromise<Void>?)
```

Comment:
Should we default the promise to nil? (Feel free to disregard if that would be inconsistent with the rest of NIO.)

Reply:
We can't provide a default as it would make `cancel()` ambiguous (and would be inconsistent with NIO).